adr/0012-extra-security-for-support.md

```diff
@@ -0,0 +1,42 @@
+# 12. Improve security of the support interface
+
+Date: 2020-03-30
+
+## Context
+
+Members of the Apply team have access to a section of the service called [Support](https://www.apply-for-teacher-training.education.gov.uk/support). It exposes a lot of data and allows make changes to the site. We use DfE Signin for access to the support interface.
+
+There are 2 concerns around the security of Support:
+
+- DfE Signin uses a username and password for sign in. It doesn’t offer 2 factor authentication. This means that an attacker could steal or guess a password and gain access to Support.
+- We do not have an automated leavers process that revokes access to DfE Signin when people leave. This leaves open the possibility that former staff retain access to Support.
+
+As part of proactive security work we’ve looked into improving the security of Support.
+
+## Decision
+
+We allow access only with `@digital.education.gov.uk` and `@education.gov.uk` addresses to Support. These email systems both have 2FA enabled. This means that if the user can prove that they have access to their email, we can be reasonably sure that they are who they say they are.
+
+The solution we intend to build works by requiring Support users to confirm that they have access to their email every week, on each machine they use to access the service. This also proves that they’re still employed by DfE.
+
+The technical implementation:
+
+When a support user visits the site they sign in as usual using DfE Signin. When they return to Apply, we check if they have a valid “email\_confirmed” cookie set. If not, they’ll see a screen saying that they’ve received an email with a confirmation link. If they click the link, we’ll set the “email\_confirmed” cookie and allow them access to the site.
+
+The “email\_confirmed” cookie is [signed to avoid tampering](https://apidock.com/rails/v6.0.0/ActionDispatch/Cookies/ChainedCookieJars/signed). It contains the user ID of the user (so it can’t be shared) and an expiry date 7 days in the future (so that emails have to be reconfirmed every 7 days).
+
+## Other options that have been considered
+
+For this ADR we considered a wide variety of options. In addition to solving the 2 issues described above, we looked at how straightforward the implementation would be, the ease of use, the level of security it would provide, and if it could be re-used to provide additional security for the provider interface.
+
+All options have their trade-offs. We’ve chosen to go with the Email confirmation option, since it is fairly straightforward to implement and it fulfills our 2 requirements (unlike B, C, D, E, F). Compared to option G (SSO), it’s easier to migrate to as it doesn’t change as much for users and works for all users.
```

Comment:
```suggestion
All options have their trade-offs. We’ve chosen to go with the Email confirmation option, since it is fairly straightforward to implement and it fulfills our 2 requirements (unlike B, C, D, E, F). Option G (SSO) is not possible as we need to support users on the `@education.gov.uk` domain which is not a Google apps domain.
```